Update moose-environment stack

milljm opened the pull request:
Update all recipes, so we have a new build as provided by moose/conda recipes.

Update compilers. And in doing so, all packages which depend on it.
MPICH: To use new compilers package.
PETSc: 3.12.5, to new use MPICH package.
libMesh: to use new PETSc package. Add oprof method.

Closes #15058

Commit:
Update moose-environment stack

Update all recipes, so we have a new build as provided by moose/conda recipes.

Update compilers. And in doing so, all packages which depend on it.
MPICH: To use new compilers package.
PETSc: 3.12.5, to new use MPICH package.
libMesh: to use new PETSc package. Add oprof method.

Closes #15058

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/499957/) on 671f6d1 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/15059/site)

This comment will be updated on new commits.

Commit:
Shift dependencies

There is some sort of new conflict with Mac, and the way I am setting
dependencies for moose-compilers/moose-mpich. Instead of having moose-mpich
depend on moose-compilers, have both contain the exact same set of dependencies
by created a symlink between the two conda_build.config.yaml files.

Why this works, but not when they depend on each other, is a mystery to me at
the moment.

Shift the libpng dependency into libMesh. As it makes a bit more sense having
that there, rather than in compilers/mpich. It may even belong in moose-tools,
but moose-tools is not required to perform run_tests, though, libpng is.

Refs #15058

Commit:
Do not attempt to use libcxx 10.x

Received an error while building libMesh. This was one of the only major changes,
so revert the version back to what we know works.

Refs #15058

permcody reviewed (approved):
I don't really know much about any of this. I'm happy to merge!
